bug(prefer-timer-args): unbound method breaks `this` in classes

`e18e/prefer-timer-args` breaks when a method inside a `setInterval`/`setTimeout` attempts to access the class instance using `this.`.

It also collides with [@typescript-eslint/unbound-method](https://typescript-eslint.io/rules/unbound-method), which attempts to prevent bugs like this.
```
  11:21  error  A method that is not declared with `this: void` may cause unintentional scoping of `this` when separated from its object.
Consider using an arrow function or explicitly `.bind()`ing the method to avoid calling the method with an unintended `this` value.
If a function does not access `this`, it can be annotated with `this: void`  @typescript-eslint/unbound-method
```

## Before: Working
<img width="661" height="543" alt="Image" src="https://github.com/user-attachments/assets/54ba0d60-fdfe-4889-a11a-e71d3ecf3ee0" />

## After: Broken
<img width="1176" height="820" alt="Image" src="https://github.com/user-attachments/assets/b27166ea-aab1-47d0-a83d-8db8d8543920" />

--- a/src/rules/prefer-timer-args.test.ts
+++ b/src/rules/prefer-timer-args.test.ts
@@ -74,7 +74,16 @@ ruleTester.run('prefer-timer-args', preferTimerArgs, {
     // Edge cases
     'setTimeout()',
     'setTimeout(() => fn())',
-    'notSetTimeout(() => doSomething(a), 100)'
+    'notSetTimeout(() => doSomething(a), 100)',
+
+    // References to `this` should not be transformed (would lose binding)
+    'setTimeout(() => this.foo(), 100)',
+    'setTimeout(() => this.handler.process(data), 500)',
+    'setTimeout(() => fn(this.value), 100)',
+    'setTimeout(() => fn(a, this.b, c), 100)',
+    'setInterval(() => this.tick(), 1000)',
+    'window.setTimeout(() => this.render(), 16)',
+    'setTimeout(this.method.bind(null, arg), 100)'
   ],
 
   invalid: [
@@ -126,17 +135,6 @@ ruleTester.run('prefer-timer-args', preferTimerArgs, {
     {
       code: 'setTimeout(() => obj.method(arg), 100)',
       output: 'setTimeout(obj.method, 100, arg)',
-      errors: [
-        {
-          messageId: 'preferArgs'
-        }
-      ]
-    },
-
-    // Nested property access
-    {
-      code: 'setTimeout(() => this.handler.process(data), 500)',
-      output: 'setTimeout(this.handler.process, 500, data)',
       errors: [
         {
           messageId: 'preferArgs'

--- a/src/rules/prefer-timer-args.ts
+++ b/src/rules/prefer-timer-args.ts
@@ -13,6 +13,27 @@ function isNullOrUndefined(node: Expression): boolean {
   return node.type === 'Identifier' && node.name === 'undefined';
 }
 
+function isTimerCall(node: CallExpression): boolean {
+  if (
+    node.callee.type === 'Identifier' &&
+    (node.callee.name === 'setTimeout' || node.callee.name === 'setInterval')
+  ) {
+    return true;
+  }
+  if (
+    node.callee.type === 'MemberExpression' &&
+    node.callee.object.type === 'Identifier' &&
+    (node.callee.object.name === 'window' ||
+      node.callee.object.name === 'globalThis') &&
+    node.callee.property.type === 'Identifier' &&
+    (node.callee.property.name === 'setTimeout' ||
+      node.callee.property.name === 'setInterval')
+  ) {
+    return true;
+  }
+  return false;
+}
+
 function isSafeArgument(arg: Expression | SpreadElement): boolean {
   if (arg.type === 'SpreadElement') {
     return arg.argument.type === 'Identifier';
@@ -101,23 +122,26 @@ export const preferTimerArgs: Rule.RuleModule = {
   },
   create(context) {
     const sourceCode = context.sourceCode;
+    let sawThis = false;
 
     return {
-      CallExpression(node: CallExpression) {
-        // Check if this is setTimeout/setInterval (with optional window/globalThis prefix)
-        const isTimerFunction =
-          (node.callee.type === 'Identifier' &&
-            (node.callee.name === 'setTimeout' ||
-              node.callee.name === 'setInterval')) ||
-          (node.callee.type === 'MemberExpression' &&
-            node.callee.object.type === 'Identifier' &&
-            (node.callee.object.name === 'window' ||
-              node.callee.object.name === 'globalThis') &&
-            node.callee.property.type === 'Identifier' &&
-            (node.callee.property.name === 'setTimeout' ||
-              node.callee.property.name === 'setInterval'));
+      ThisExpression() {
+        sawThis = true;
+      },
 
-        if (!isTimerFunction) {
+      CallExpression(node: CallExpression) {
+        if (isTimerCall(node)) {
+          sawThis = false;
+        }
+      },
+
+      'CallExpression:exit'(node: CallExpression) {
+        if (!isTimerCall(node)) {
+          return;
+        }
+
+        // Skip if the callback references `this` - transformation would lose binding
+        if (sawThis) {
           return;
         }
 

